pkg/network/infraconfigurators/bridge.go

```diff
@@ -85,18 +99,30 @@ func (b *BridgePodNetworkConfigurator) GenerateNonRecoverableDHCPConfig() *cache
 
 	dhcpConfig := &cache.DHCPConfig{
 		MAC:          *b.vmMac,
-		IPAMDisabled: !b.ipamEnabled,
-		IP:           b.podIfaceIP,
+		IPAMDisabled: false,
 	}
 
-	if b.ipamEnabled && len(b.podIfaceRoutes) > 0 {
-		log.Log.V(4).Infof("got to add %d routes to the DhcpConfig", len(b.podIfaceRoutes))
-		b.decorateDhcpConfigRoutes(dhcpConfig)
+	if b.podIfaceIP != nil {
+		dhcpConfig.IP = *b.podIfaceIP
+		if len(b.podIfaceRoutes) > 0 {
+			log.Log.V(4).Infof("got to add %d routes to the DhcpConfig", len(b.podIfaceRoutes))
+			b.decorateDhcpConfigRoutes(dhcpConfig)
+		}
+	}
+	if b.podIfaceIPv6 != nil {
+		dhcpConfig.IPv6 = *b.podIfaceIPv6
 	}
+
 	return dhcpConfig
 }
 
 func (b *BridgePodNetworkConfigurator) PreparePodNetworkInterface() error {
+	// Ensure that any IPv6 address will be removed by setting the link down
+	if err := b.handler.ConfigureIpv6FlushAddrOnDown(); err != nil {
+		log.Log.Reason(err).Error("failed to set keep_addr_on_down=-1")
```

Comment:
would we want to print out the sysctl error ? Would that info be helpful ?

Reply:
I assumed that `.Reason(err)` would already do that. In any case, I based this on the existing ARP code for IPv4, so if we change logging here we'll probably also want to change it there.

Reply:
fair enough.